test/ModuleInterface/type_wrappers.swift

```diff
@@ -0,0 +1,123 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-emit-module-interface(%t/TypeWrappers.swiftinterface) %s -module-name TypeWrappers -enable-experimental-feature TypeWrappers
+// RUN: %target-swift-typecheck-module-from-interface(%t/TypeWrappers.swiftinterface) -module-name TypeWrappers
+// RUN: %FileCheck %s < %t/TypeWrappers.swiftinterface
+
+// CHECK: @typeWrapper public struct Wrapper<W, S> {
+// CHECK-NEXT: public init(for: W.Type, storage: S)
+// CHECK-NEXT:   public subscript<V>(propertyKeyPath _: Swift.KeyPath<W, V>, storageKeyPath path: Swift.KeyPath<S, V>) -> V {
+// CHECK-NEXT:     get
+// CHECK-NEXT:   }
+// CHECK-NEXT:   public subscript<V>(propertyKeyPath _: Swift.KeyPath<W, V>, storageKeyPath path: Swift.WritableKeyPath<S, V>) -> V {
+// CHECK-NEXT:     get
+// CHECK-NEXT:     set
+// CHECK-NEXT:   }
+// CHECK-NEXT: }
+
+@typeWrapper
+public struct Wrapper<W, S> {
+  public init(for: W.Type, storage: S) {}
+
+  public subscript<V>(propertyKeyPath _: KeyPath<W, V>, storageKeyPath path: KeyPath<S, V>) -> V {
+    get { fatalError() }
+  }
+
+  public subscript<V>(propertyKeyPath _: KeyPath<W, V>, storageKeyPath path: WritableKeyPath<S, V>) -> V {
+    get { fatalError() }
+    set { }
+  }
+}
+
+// CHECK: @TypeWrappers.Wrapper public class Test<T> where T : Swift.StringProtocol {
+// CHECK:   public init(a: Swift.Int, b: [T])
+// CHECK:   public init(storageWrapper: TypeWrappers.Wrapper<TypeWrappers.Test<T>, TypeWrappers.Test<T>.$Storage>)
+// CHECK: }
+
+@Wrapper
+public class Test<T: StringProtocol> {
+  var a: Int
+  let b: [T]
+}
+
+@Wrapper
+public protocol Wrapped {
+  init()
+}
+
+public protocol OuterWrapped : Wrapped {}
+
+// CHECK: public struct WithProtocol : TypeWrappers.Wrapped {
+// CHECK:   public var a: Swift.Int {
+// CHECK:     get
+// CHECK:     set
+// CHECK:   }
+// CHECK:   public var b: Swift.String {
+// CHECK:     get
+// CHECK:     set
+// CHECK:   }
+// CHECK:   public init()
+// CHECK:   public var $storage: TypeWrappers.Wrapper<TypeWrappers.WithProtocol, TypeWrappers.WithProtocol.$Storage>
+// CHECK:   public struct $Storage {
+// CHECK:   }
+// CHECK:   public init(storageWrapper: TypeWrappers.Wrapper<TypeWrappers.WithProtocol, TypeWrappers.WithProtocol.$Storage>)
+// CHECK: }
+
+public struct WithProtocol : Wrapped {
+  public var a: Int
+  public var b: String
+
+  public init() {
+  }
+}
+
+// CHECK: @TypeWrappers.Wrapper final public class ClassWithProtoocol : TypeWrappers.Wrapped {
+// CHECK:   required public init()
+// CHECK:   final public var $storage: TypeWrappers.Wrapper<TypeWrappers.ClassWithProtoocol, TypeWrappers.ClassWithProtoocol.$Storage>
+// CHECK:   public struct $Storage {
+// CHECK:   }
+// CHECK:   public init(storageWrapper: TypeWrappers.Wrapper<TypeWrappers.ClassWithProtoocol, TypeWrappers.ClassWithProtoocol.$Storage>)
+// CHECK: }
+
+@Wrapper
```

Comment:
Is the explicit `@Wrapper` intentional here? I thought this test case might be testing the same thing as above, but with a `class` instead of a `struct`.

Reply:
It is intentional to make sure that there is no clash there.

Reply:
I actually realized that there is no attribute here although I think there should be, I'm currently working on fixing that.

Reply:
Fixed!